lib/std/system/stringimpl.nim

```diff
@@ -70,30 +84,35 @@ proc `=dup`*(s: string): string {.exportc: "nimStrDup", inline, nodestroy.} =
   else:
     result = s
 
+# --- string/cstring conversions ---
+
+func strlen(a: cstring): csize_t {.importc: "strlen", header: "<string.h>".}
+
 proc borrowCStringUnsafe*(s: cstring; len: int): string =
   ## Creates a Nim string from a `cstring` by borrowing the
   ## underlying storage. You have to ensure the `cstring` lives
   ## long enough for this to be safe! If in doubt,
   ## use `fromCString` instead.
   string(a: cast[StrData](s), i: (len shl LenShift))
 
-func strlen(a: cstring): csize_t {.importc: "strlen", header: "<string.h>".}
-
 proc borrowCStringUnsafe*(s: cstring): string =
   ## Creates a Nim string from a `cstring` by borrowing the
   ## underlying storage. You have to ensure the `cstring` lives
   ## long enough for this to be safe! If in doubt,
   ## use `fromCString` instead.
   string(a: cast[StrData](s), i: (int(strlen(s)) shl LenShift))
 
+proc isTerminatingZero*(s: string): bool {.inline.} =
```

Comment:
1. Name should be `hasTerminatingZero`.
2. This proc sounds like a bad idea as it introduces non-determinism for allocations that don't initialize which we generally prefer these days.

Reply:
removing it makes sense despite i can implement it somewhere in my code thanks to `rawData`